### PR TITLE
feature: http status message enum

### PR DESCRIPTION
**Pull request by md-redwan-hossain**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## What is the current behavior?

NestJS provides a useful HttpStatus enum for sending response codes manually. However, when creating custom exception filters, it's often necessary to replicate the default error response structure, including the string representation of the HTTP status code. Writing the string representation can be error prone regarding grammar. Unfortunately, NestJS doesn't currently offer such an enum.

Issue Number: N/A

## What is the new behavior?
This commit will allow to send the string representation of the HTTP status code using `HttpStatusMessage` enum. Also, it can be used wherever string values are needed for HTTP status codes.

```typescript
const errorBody = {
  message: exception.message,
  error: HttpStatusMessage.UNAUTHORIZED,
  statusCode: HttpStatus.UNAUTHORIZED
};
```
The result will be following:

```json
{
  "message": "jwt expired",
  "error": "Unauthorized",
  "statusCode": "401"
}
```

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

## Other information

### CODE REVIEW

File: packages/common/enums/http-status-message.enum.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
+/**
+ * @publicApi
+ */
+export enum HttpStatusMessage {
```
Comment (dammy001) [inline packages/common/enums/http-status-message.enum.ts:4]: I feel this should be done on user land and not the framework. It depends on how you choose to send the response payload. It's not totally required to send status message.

Comment (md-redwan-hossain) [reply packages/common/enums/http-status-message.enum.ts:4]: The enum does not format the response payload message in any way; it simply serves as the string representation of the values within the `HttpStatus` enum of Nest. You can use it wherever string values are needed for HTTP status codes.

**Review by thaislour — APPROVED** [review]

File: packages/common/enums/http-status-message.enum.ts (hunk 1 above)
Comment (micalevisk) [reply packages/common/enums/http-status-message.enum.ts:4]: I agree with @dammy001 

no reason to add such enum here as it is not used by the framework but the user-land. Let's keep the codebase clean as possible we can so we won't ship dead code.

Comment (md-redwan-hossain) [reply packages/common/enums/http-status-message.enum.ts:4]: I cross-checked both enums manually and with AI tools. The enum keys are identical in my inspection. As for dead code, yes, this logic seems legitimate. However, a batteries-included framework should enhance the developer experience by providing as many prebuilt features as possible. Furthermore, Nest, by default, sends the string representation of the HTTP status code in the error response; this enum can be beneficial in this context.

**Comment by kamilmysliwiec** [comment]
Thanks for your suggestion!

There are no plans to implement it in the foreseeable future.

If you think your request could live outside Nest's scope, we'd encourage you to collaborate with the [community](https://discord.gg/G7Qnnhy) on publishing it as an open source package.

**Pull request closed without merging**